airflow/providers/amazon/aws/secrets/systems_manager.py

```diff
@@ -166,16 +185,26 @@ def get_config(self, key: str) -> str | None:
         if self.config_prefix is None:
             return None
 
-        return self._get_secret(self.config_prefix, key)
+        return self._get_secret(self.config_prefix, key, self.config_lookup_pattern)
 
-    def _get_secret(self, path_prefix: str, secret_id: str) -> str | None:
+    def _get_secret(self, path_prefix: str, secret_id: str, lookup_pattern: str | None) -> str | None:
         """
         Get secret value from Parameter Store.
 
         :param path_prefix: Prefix for the Path to get Secret
         :param secret_id: Secret Key
+        :param lookup_pattern: If provided, `secret_id` must match this pattern to look up the secret in
+            Systems Manager
         """
+        if lookup_pattern is not None and not re.match(lookup_pattern, secret_id, re.IGNORECASE):
+            return None
+
         ssm_path = self.build_path(path_prefix, secret_id)
+
+        # AWS Systems Manager mandate to have a leading "/". Adding it dynamically if not there
+        if not ssm_path.startswith("/"):
+            ssm_path = f"/{ssm_path}"
```

Comment:
Maybe break this out into a function? I know it's a small nit, but would be a bit cleaner.